homeassistant/components/stream/__init__.py

```diff
@@ -0,0 +1,156 @@
+"""
+Provide functionality to stream video source.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/stream/
+"""
+import logging
+import threading
+
+import voluptuous as vol
+
+from homeassistant.auth.util import generate_secret
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.loader import bind_hass
+
+from .const import (
+    DOMAIN, CONF_KEEPALIVE, ATTR_STREAMS, ATTR_ENDPOINTS)
+from .worker import stream_worker
+from .hls import async_setup_hls
+
+REQUIREMENTS = ['av==6.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_KEEPALIVE, default=False): cv.boolean,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+# pylint: disable=dangerous-default-value
+@bind_hass
+async def async_request_stream(hass, stream_source, options={}, preload=False):
+    """Set up stream with token."""
+    if DOMAIN not in hass.config.components:
+        raise HomeAssistantError("Stream is not configured.")
+
+    if not stream_source:
+        raise HomeAssistantError("Invalid stream source.")
+
+    keepalive = hass.data[DOMAIN][CONF_KEEPALIVE]
+    try:
+        streams = hass.data[DOMAIN][ATTR_STREAMS]
+        stream = streams.get(stream_source)
+        if not stream:
+            stream = Stream(hass, stream_source,
+                            options=options, keepalive=keepalive)
```

Comment:
I think that we should not expose the options object directly, but instead have these things be set via keyworkd args.

```python
async def async_request_stream(hass, stream_source, *, access_token=None, preload=False):
```